tests/unit_tests/gui/model/test_snapshot.py

```diff
@@ -39,22 +47,107 @@ def test_realization_sort_order(full_snapshot):
     for i in range(0, 100):
         iter_index = model.index(i, 0, model.index(0, 0, QModelIndex()))
 
-        assert str(i) == iter_index.internalPointer().id_, print(
+        assert str(i) == str(iter_index.internalPointer().id_), print(
             i, iter_index.internalPointer()
         )
 
 
-def test_realization_job_hint(full_snapshot):
+def test_realization_forward_model_colors(full_snapshot):
     model = SnapshotModel()
     model._add_snapshot(SnapshotModel.prerender(full_snapshot), 0)
 
     partial = PartialSnapshot(full_snapshot)
     partial.update_forward_model(
-        "0", "0", ForwardModel(status=FORWARD_MODEL_STATE_RUNNING)
+        "0", "0", ForwardModel(status=FORWARD_MODEL_STATE_FINISHED)
+    )
+    partial.update_forward_model(
+        "0", "1", ForwardModel(status=FORWARD_MODEL_STATE_FAILURE)
+    )
+    partial.update_forward_model(
+        "0", "2", ForwardModel(status=FORWARD_MODEL_STATE_RUNNING)
     )
     model._add_partial_snapshot(SnapshotModel.prerender(partial), 0)
+    first_real = model.index(0, 0, model.index(0, 0))
+
+    queue_color = model.data(first_real, RealStatusColorHint)
+    assert queue_color.name() == QColor(*COLOR_RUNNING).name()
+
+    colors = model.data(first_real, RealJobColorHint)
+    color_list = [
+        QColor(*COLOR_FINISHED).name(),
+        QColor(*COLOR_FAILED).name(),
+        QColor(*COLOR_RUNNING).name(),
+        QColor(*COLOR_PENDING).name(),
+    ]
+    status_list = [
+        FORWARD_MODEL_STATE_FINISHED,
+        FORWARD_MODEL_STATE_FAILURE,
+        FORWARD_MODEL_STATE_RUNNING,
+        FORWARD_MODEL_STATE_START,
+    ]
+
+    for i in range(4):
+        assert colors[i].name() == color_list[i]
+
+        # verify forward_model coloring and status texts
+        status = model.data(model.index(i, 2, first_real), Qt.DisplayRole)
+        color = model.data(model.index(i, 2, first_real), Qt.BackgroundRole)
+        assert status == status_list[i]
+        assert color.name() == color_list[i]
+
+
+def test_realization_queue_wait_color_if_none_running(waiting_snapshot):
```

Comment:
I thought to be more explicit with the function name, but not sure tbh `test_that_realization_has_wait_color_when_none_is_running`? :)